fragment_reader.go

```diff
@@ -0,0 +1,83 @@
+package uploader
+
+import (
+	"io"
+)
+
+// fragmentReader implements both Reader and ByteReader
```

Comment:
Where's `ReadByte` implemented?

Reply:
ah I removed that bit of the implementation. I'll update the comment. It was useful before I understood that I couldn't use `gzip`'s `Reader` to compress and needed to use its `Writer` interface.

Reply:
69c0e3776af50afcce139ae280b2270a55e341b9